### PR TITLE
FEAT: Added a new property to check if a polygon has self.intersections and  a method to remove it from the polygon.

### PR DESCRIPTION
**Pull request by maxcapodi78**
Added also a method in layout_validation.py to remove nets self intersections.

### CODE REVIEW

File: src/pyedb/dotnet/edb_core/layout_validation.py [hunk 1]
```diff
@@ -258,6 +258,34 @@ def area_calc(elem):
...
+        _objects_list = {}
+        _padstacks_list = {}
+        new_prims = []
+        for prim in self._pedb.modeler.polygons:
+            if prim.net_name in net_list:
+                new_prims.extend(prim.remove_self_intersections())
+        if new_prims:
+            self._pedb._logger.info("Self-Intersections detected and removed.")
+        else:
+            self._pedb._logger.info("No Self-Intersections found.")
```
Comment (SMoraisAnsys) [inline src/pyedb/dotnet/edb_core/layout_validation.py:286]:
```suggestion
            self._pedb._logger.info("Self-intersection not found.")
```

**Review by SMoraisAnsys — CHANGES_REQUESTED** [review]
Great job ! I just left a couple of comments.

File: src/pyedb/dotnet/edb_core/layout_validation.py [hunk 2]
```diff
@@ -258,6 +258,34 @@ def area_calc(elem):
 
         return new_nets
 
+    def self_intersections(self, net_list):
```
[the remark this answers is not in the document]
Comment (svandenb-dev) [reply src/pyedb/dotnet/edb_core/layout_validation.py:261]: I agree the name looks like a property

**Comment by svandenb-dev** [comment]
Thanks a lot @maxcapodi78 !!

**Review by svandenb-dev — DISMISSED** [review]
LGTM

**Review by SMoraisAnsys — APPROVED** [review]
LGTM, great job again !

**Review by svandenb-dev — APPROVED** [review]
LGTM

**Pull request merged**